Add PHP autofix to pre-commit hook

Aljullu opened the pull request:
Fixes #2021.

Add PHP Lint autofix to pre-commit hook.

### Detailed test instructions:
- Edit any PHP file and add something like this:
```PHP
if ( 1 == 1 ) {
   return; // Spaces instead of tabs
}
```
- With git, add that file and do a commit.
- Notice the spaces instead of tabs issue if automatically fixed.
- Notice not adding a period in the comment prevents you from committing and you manually have to fix that issue.

Aljullu committed:
Add PHP autofix to pre-commit hook

psealock reviewed (commented):
I get a different message when committing your code snippet

<img width="469" alt="Screen Shot 2019-04-10 at 9 53 08 AM" src="https://user-images.githubusercontent.com/1922453/55837860-abe9b700-5b76-11e9-9c13-15e1f576e890.png">

When I remove the comment, the commit is allowed to happen. `npm run lint:php` does catch the errors.

<img width="713" alt="Screen Shot 2019-04-10 at 9 54 59 AM" src="https://user-images.githubusercontent.com/1922453/55837955-fb2fe780-5b76-11e9-9208-e4decca23808.png">

Is this not what you're experiencing?

Aljullu committed:
Remove unnecesary 'dev/null' output path

Aljullu committed:
Add error message if PHP_CodeSniffer is not installed

Aljullu committed:
Add periods to comments

Aljullu committed:
Fix comment

Aljullu commented:
Thanks for the review @psealock! You're right, my example was wrong. Strict comparisons is a warning instead of an error, so it doesn't block commiting. I think that's to be expected and we are currently not catching warnings either.

I updated the steps to reproduce and added some code to check if PHP_CodeSniffer is installed. Feel free to take another look. :+1:

joshuatf commented on bin/pre-commit-hook.js:
```diff
@@ -91,30 +91,55 @@ if ( lintResult.status ) {
+		execSync( `./vendor/bin/phpcs -h` );
+	} catch( e ) {
+		console.log(
+			'PHP_CodeSniffer is not installed. ' +
```
❤️

joshuatf reviewed (approved):
Sweet.  Testing well for fixable phpcbf items.  LGTM if @psealock doesn't have any other concerns.

psealock reviewed (approved):
Oh right, I didn't see those were warnings. LGTM
